aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -95,20 +95,21 @@ void TensorIterator::compute_types() {
   if (missing_dtypes || compute_common_dtype_) {
     auto& type = compute_common_type();
     for (auto& op : operands_) {
+      auto& op_tensor_type = at::globalContext().getNonVariableType(op.tensor.type().backend(), op.tensor.type().scalarType());
```

Comment:
this file seems a bit different from the others (which look correct), because here the checks should all still be valid, i.e. they are checking device type, scalarType, etc. which are all still valid on scalars.  What's the symptom here?